app/controllers/posts_controller.rb

```diff
@@ -1,6 +1,15 @@
 class PostsController < ApplicationController
   before_action :authenticate_user!, except: :index
-  before_action :require_permission, except: :create
+  before_action :require_permission, except: [:create, :repost]
```

Comment:
The except list is longer than the only list would be for this before_action, and the before_action in blogs_controller. This might just be for my benefit, but my logic was to force me to think about it on every action/feature that could be added in the future.